src/JsonSchema/Constraints/ObjectConstraint.php

```diff
@@ -170,6 +175,26 @@ protected function &getProperty(&$element, $property, $fallback = null)
     }
 
     /**
+     * set and return a property on an object or array
+     *
+     * @param mixed $element Element to manipulate
+     * @param string $property  Property to set
+     * @param mixed $value Value to set
+     */
+    protected function &setProperty(&$element, $property, $value)
```

Comment:
I guess there's no law against it, but having a setter return something just feels like an unnecessary mixing of concerns to me. Throwing in the reference is just extra squirm factor. You're only calling this method in one place, and though it's late and my head is fuzzy I really can't see why it's necessary or how its inclusion makes the code any simpler.

```
if ($default && isset($definition->default) && $property === $undefinedConstraint) {
    $property = $definition->default;
    $this->setProperty($element, $i, $property);
}
```